### PR TITLE
Added the tutorial "Find in buffer" (1979)

### PR DESCRIPTION
**Pull request by muream**
Added the tutorial suggested [here](https://github.com/onivim/oni/issues/1979)

### CODE REVIEW

**Comment by bryphe** [comment]
This is awesome @muream - exactly what I had in mind for it 👍 Super cool and a great example for other people who want to write tutorials. Thank you for putting this together!

**Comment by CrossR** [comment]
Just given the tutorial a test, works well!
Going to be great to have a big suite of tutorials to learn from!

File: browser/src/Services/Learning/Tutorial/Tutorials/SearchInBufferTutorial.tsx [hunk 1]
```diff
@@ -0,0 +1,81 @@
+/**
+ * TutorialManager
```
Comment (robin-pham) [inline browser/src/Services/Learning/Tutorial/Tutorials/SearchInBufferTutorial.tsx]: I'm not super familiar with this codebase (having just started), but it seems like this comment is supposed to be  
```
/**
 * {filename}
 *
 * {short description}
 */

```

File: browser/src/Services/Learning/Tutorial/Tutorials/SearchInBufferTutorial.tsx [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+ */
+
+import * as React from "react"
+
+import { ITutorial, ITutorialMetadata, ITutorialStage } from "./../ITutorial"
+// import { InitializeBufferStage, MoveToGoalStage } from "./../Stages"
```
Comment (robin-pham) [inline browser/src/Services/Learning/Tutorial/Tutorials/SearchInBufferTutorial.tsx]: I think we can just remove this line if it's not used.

**Comment by bryphe** [comment]
I'll bring this in since I'm wrapping up things for the 0.3.2 release - want to make sure this makes it in! 👍

File: browser/src/Services/Learning/Tutorial/Tutorials/SearchInBufferTutorial.tsx (hunk 1 above)
Comment (bryphe) [reply browser/src/Services/Learning/Tutorial/Tutorials/SearchInBufferTutorial.tsx]: Ah sorry I missed this @robin-pham - yep, that's the desired format for sure! 

I think unfortunately @muream inherited my mistake, some of the other tutorials I wrote have this problem too... sorry @muream!

File: browser/src/Services/Learning/Tutorial/Tutorials/SearchInBufferTutorial.tsx (hunk 2 above)
Comment (bryphe) [reply browser/src/Services/Learning/Tutorial/Tutorials/SearchInBufferTutorial.tsx]: Shoot, another case where I think @muream inherited my mistake (I see this in `BasicMovementTutorial.tsx` too).

**Comment by bryphe** [comment]
Thanks for the PR and work on this @muream - great job! I always forget to use `?` so maybe this will help me practice.

File: browser/src/Services/Learning/Tutorial/Tutorials/SearchInBufferTutorial.tsx (hunk 1 above)
Comment (muream) [reply browser/src/Services/Learning/Tutorial/Tutorials/SearchInBufferTutorial.tsx]: Haha, yep, I straight up copied the basic movement tutorial and built up from there.
I'll make sure to check for this kind of things going forward!

**Pull request merged**